pkg/runtime/depot.go

```diff
@@ -253,10 +278,28 @@ func (d *depot) DeployViaCopy(id strfmt.UUID, relativeSrc, absoluteDest string)
 		Files:       files.RelativePaths(),
 		RelativeSrc: relativeSrc,
 	})
+	d.recordUse(id)
 
 	return nil
 }
 
+func (d *depot) recordUse(id strfmt.UUID) {
+	// Ensure a cache entry for this artifact exists and then update its last access time.
+	if _, exists := d.config.Cache[id]; !exists {
+		size, err := fileutils.GetDirSize(d.Path(id))
+		if err != nil {
+			multilog.Error("Could not get artifact size on disk: %v", err)
```

Comment:
I'd rather we return it. If this fails something is definitely wrong cause we own the depot. Realistically this should pretty much never fail.